### PR TITLE
Recall

### PR DESCRIPTION
**Pull request by BlckHawker**
[Ticket](https://github.com/orgs/RIT-Crowd-Comic/projects/2?pane=issue&itemId=63996328)

### Description

Use `/recall` to outputs a JSON of all the remembered texts

Returns the following data from each message:
- Author's id
- Author's name
- message content
- message id
- timestamp (in UTC)

#### Parameters
- N/A

#### Screenshot(s)
Screenshot 2024-05-28 130044

### Known Issues / Warnings
- All messages are currently being stored in `rememberedMessages.json`. This does not seem secure nor practical. I assume there is a way to store a JSON without the use of a file, but I assume it would be obsolete once the database is implemented.
- I assume there is a way to format the JSON, so it's not one line, but unsure how to do so

### Possible Additional Features
- Other meta data variables can be added or removed from the message object. Would affect other remember commands, though.

### CODE REVIEW

**Review by wct9667 — APPROVED** [review]
Looks good to me.

File: src/commands/remember/recall.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            //get the json
+            //? There is probably a way to do this so we don't have to have a fle created to save to
+            //? Will probably be resolved once DB is created
+            await interaction.deferReply({ ephemeral: false })
+            const jsonFilePath = './src/rememberedMessages.json';
+            const messageObj = []
+            const messages = getRememberedMessage();
+            messages.forEach(m => messageObj.push(m))
+            interaction.editReply("Success")
+            const json = JSON.stringify(messageObj)
```
Comment (tstodter) [inline src/commands/remember/recall.js:21]: `JSON.stringify(messageObj, null, 2)` should get you the formatting you want.

File: src/commands/remember/recall.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        try {
+            //get the json
+            //? There is probably a way to do this so we don't have to have a fle created to save to
+            //? Will probably be resolved once DB is created
+            await interaction.deferReply({ ephemeral: false })
+            const jsonFilePath = './src/rememberedMessages.json';
+            const messageObj = []
+            const messages = getRememberedMessage();
+            messages.forEach(m => messageObj.push(m))
+            interaction.editReply("Success")
```
Comment (tstodter) [inline src/commands/remember/recall.js:20]: Doesn't look like you actually need `messageObj` here; you're just shallow cloning `messages`, so you could probably just do `const json = JSON.stringify(messages, null, 2)` instead.

File: src/commands/remember/recall.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const { ApplicationCommandOptionType, Attachment } = require('discord.js');
+const { getRememberedMessage } = require("../../utils/rememberMessages");
+const { ephemeral } = require('../../../config.json');
+var fs = require('fs');
```
Comment (tstodter) [inline src/commands/remember/recall.js:4]: No need for `var` here, just use `const`.

**Pull request merged**